algorithms/src/sorting/Kokkos_Sort.hpp

```diff
@@ -0,0 +1,196 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_SORT_FREE_FUNCTION_HPP_
+#define KOKKOS_SORT_FREE_FUNCTION_HPP_
+
+#include "impl/Kokkos_SortFreeFunction.hpp"
+#include <Kokkos_Core.hpp>
+#include <std_algorithms/Kokkos_BeginEnd.hpp>
+#include <std_algorithms/Kokkos_Copy.hpp>
+#include <Kokkos_DynamicView.hpp>
+#include <algorithm>
+
+namespace Kokkos {
+// clang-format off
+
+//
+// basic overloads
+//
+template <class ExecutionSpace, class DataType, class... Properties>
+void sort([[maybe_unused]] const ExecutionSpace& exec,
+          const Kokkos::View<DataType, Properties...>& view)
+{
+  // constraints
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
+  static_assert( (ViewType::rank == 1),
+		 "Kokkos::sort: currently supporting 1D Views only.");
+  static_assert(SpaceAccessibility<ExecutionSpace, MemSpace>::accessible,
+		"Kokkos::sort: execution space instance is not able to access the memory space of the "
+		"View argument!");
+
+  // trivial cases
+  if (view.extent(0) <= 1) { return; }
+
+  // non trivial
+  if constexpr (SpaceAccessibility<HostSpace, MemSpace>::accessible) {
+    auto first = ::Kokkos::Experimental::begin(view);
+    auto last  = ::Kokkos::Experimental::end(view);
+    std::sort(first, last);
```

Comment:
Should std::sort be used only when ExecutionSpace == Serial? ExecutionSpace could be host parallel but std::sort with no execution policy will run sequentially, AFAIK